### PR TITLE
Support more operators

### PR DESCRIPTION
**Pull request by Chestshire**
Changes:
1. Fixed an issue that empty set will be written as `{}` in python. Changed to `set()`.
2. Support operations between Relations.
3. Support operations between `Sig_A -> Sig_B` and Relations.
4. Support Relations in guard conditions (evaluations).
5. Support both `OR` and `AND` operators between expressions in guard conditions.
6. Support cartesian product between signatures. Used symbol `*` as a cartesian product operator.
7. Support basic expressions with quantifiers (`all` and `one`). Basic means simple in this context. 
   - That is, only one quantified variable, i.e., the variable `x` in `forall x in ...`
   - Only one quantified expression, i.e., the expression `expr` in `... | <expr>`
8. Fix assignments, assignments should only modify the values of the variables, instead of changing the variables. This is done by using the `__setattr__()` of `VariableTracker`.

### CODE REVIEW

**Comment by Chestshire** [comment]
Spent an hour trying to figure out what the build error was all about. Gave up for now. This branch does not originate from the current `rapiddash' head. Will try again when Aidan fixes the build errors in the current head of branch `rapiddash'.

**Comment by Simonliuwaterloo** [comment]
> Spent an hour trying to figure out what the build error was all about. Gave up for now. This branch does not originate from the current `rapiddash' head. Will try again when Aidan fixes the build errors in the current head of branch `rapiddash'.

Actually the issue might be in my code, taking a look now.

File: org.alloytools.alloy.application/src/test/java/ca/uwaterloo/watform/rapidDash/DashPythonTranslationTest.java [hunk 1]
```diff
@@ -89,12 +89,9 @@ public void testStateInitAndUnnamedRelations() throws Exception {
             assertTrue(decl.contains("="));
         }
 
-        // Unamed relation should be generated.
-        assertTrue(translation.relations.contains(new DashPythonTranslation.Relation("Chair_Player", "[Chair, Player]")));
-
         assertTrue(gameState.getDecls().contains("Game_active_players = Player('active_players', 3)"));
         assertTrue(gameState.getDecls().contains("Game_active_chairs = Chair('active_chairs', 3)"));
-        assertTrue(gameState.getDecls().contains("Game_occupied = Chair_Player()"));
+        assertTrue(gameState.getDecls().contains("Game_occupied = Chair * Player"));
```
Comment (Simonliuwaterloo) [inline org.alloytools.alloy.application/src/test/java/ca/uwaterloo/watform/rapidDash/DashPythonTranslationTest.java:94]: What is the type of a cartesian product in Dash, is it a relation?

**Review by Simonliuwaterloo — APPROVED** [review]

**Comment by RuomeiYan** [comment]
Looks good to me. Right now ehealth can only take transition remove_interaction. After talking to Kent, we agreed that we would need to support env var to run other transitions in ehealth. Also for farmer's puzzle and invariants in ehealth, we would need some expression evaluation and backtracking mechanisms.

**Review by RuomeiYan — APPROVED** [review]

**Review by AidanCampbell738 — APPROVED** [review]

File: org.alloytools.alloy.application/src/test/java/ca/uwaterloo/watform/rapidDash/DashPythonTranslationTest.java (hunk 1 above)
Comment (Chestshire) [reply org.alloytools.alloy.application/src/test/java/ca/uwaterloo/watform/rapidDash/DashPythonTranslationTest.java:94]: It is of type "set".

**Pull request merged**